src/battle_move_effects.c

```diff
@@ -0,0 +1,1926 @@
+    {
+
+    },
+    [MOVE_EFFECT_ELECTRIC_TERRAIN] =
+    {
+
+    },
+    [MOVE_EFFECT_PSYCHIC_TERRAIN] =
+    {
+
+    },
+    [MOVE_EFFECT_VINE_LASH] =
+    {
+
+    },
+    [MOVE_EFFECT_WILDFIRE] =
+    {
+
+    },
+    [MOVE_EFFECT_CANNONADE] =
+    {
+
+    },
+    [MOVE_EFFECT_EFFECT_SPORE_SIDE] =
+    {
+
+    },
+    [MOVE_EFFECT_PARALYZE_SIDE] =
+    {
+
+    },
+    [MOVE_EFFECT_CONFUSE_PAY_DAY_SIDE] =
+    {
+
+    },
+    [MOVE_EFFECT_CRIT_PLUS_SIDE] =
+    {
+
+    },
+    [MOVE_EFFECT_PREVENT_ESCAPE_SIDE] =
+    {
+
+    },
+    [MOVE_EFFECT_AURORA_VEIL] =
+    {
+
+    },
+    [MOVE_EFFECT_INFATUATE_SIDE] =
+    {
+
+    },
+    [MOVE_EFFECT_RECYCLE_BERRIES] =
+    {
+
+    },
+    [MOVE_EFFECT_POISON_SIDE] =
+    {
+
+    },
+    [MOVE_EFFECT_DEFOG] =
+    {
+
+    },
+    [MOVE_EFFECT_POISON_PARALYZE_SIDE] =
+    {
+
+    },
+    [MOVE_EFFECT_HEAL_TEAM] =
+    {
+
+    },
+    [MOVE_EFFECT_SPITE] =
+    {
+
+    },
+    [MOVE_EFFECT_GRAVITY] =
+    {
+
+    },
+    [MOVE_EFFECT_VOLCALITH] =
+    {
+
+    },
+    [MOVE_EFFECT_SANDBLAST_SIDE] =
+    {
+
+    },
+    [MOVE_EFFECT_YAWN_FOE] =
+    {
+
+    },
+    [MOVE_EFFECT_LOWER_EVASIVENESS_SIDE] =
+    {
+
+    },
+    [MOVE_EFFECT_AROMATHERAPY] =
+    {
+
+    },
+    [MOVE_EFFECT_CONFUSE_SIDE] =
+    {
+
+    },
+    [MOVE_EFFECT_STEELSURGE] =
+    {
+
+    },
+    [MOVE_EFFECT_TORMENT_SIDE] =
+    {
+
+    },
+    [MOVE_EFFECT_LOWER_SPEED_2_SIDE] =
+    {
+
+    },
+    [MOVE_EFFECT_FIRE_SPIN_SIDE] =
+    {
+
+    },
+    [MOVE_EFFECT_FIXED_POWER] =
+    {
+
+    },
+};
+
+#define IF_FAIL(...) if ((result->failed = (__VA_ARGS__)))
+#define IF_SUCCEED(...) if (!(result->failed = (__VA_ARGS__)))
+
+void SetMoveEffect(u32 battler, u32 effectBattler, bool32 primary, bool32 certain)
+{
+    struct SetMoveEffectResult result = { 0 };
+    const u8 *currInstr = gBattlescriptCurrInstr;
+    SetMoveEffectWithResult(&result, battler, effectBattler, gBattleScripting.moveEffect, primary, certain, gBattlescriptCurrInstr + 1);
+
+    // Assign next instruction
+    SetMoveEffectHandleResult(&result, currInstr, gBattlescriptCurrInstr + 1);
+}
+
+// To avoid confusion the arguments are naned battler/effectBattler since they can be different from gBattlerAttacker/gBattlerTarget
+struct SetMoveEffectResult *SetMoveEffectWithResult(struct SetMoveEffectResult *result, u32 battler, u32 effectBattler, u16 moveEffect, bool32 primary, bool32 certain, const u8 *failInstr)
+{
+    s32 i;
+    bool32 affectsUser = (battler == effectBattler);
+    bool32 mirrorArmorReflected = (GetBattlerAbility(gBattlerTarget) == ABILITY_MIRROR_ARMOR);
+    union StatChangeFlags flags = {0};
+    u32 battlerAbility;
+
+    // Get battlers
+    result->battlerAtk = gBattlerAttacker;
+    result->battlerDef = gBattlerTarget;
+
+    // Set move effect and check if it exists
+    IF_FAIL((result->moveEffect = gBattleScripting.moveEffect = moveEffect) == MOVE_EFFECT_NONE)
+        return result;
```

Comment:
```suggestion
    if (moveEffect == MOVE_EFFECT_NONE)
    {
        result->moveEffect = gBattleScripting.moveEffect = moveEffect)
        result->failed = TRUE;
        return result;
    }
```
Isn't this the same?